gen/struct_test.go

```diff
@@ -729,20 +731,55 @@ func TestStructFromWireUnrecognizedField(t *testing.T) {
 			}}),
 			want: ts.ContactInfo{EmailAddress: "foo"},
 		},
+		{
+			desc: "unknown field, wrong order",
+			give: wire.NewValueStruct(wire.Struct{Fields: []wire.Field{
+				{ID: 2, Value: wire.NewValueI32(42)},
+				{ID: 1, Value: wire.NewValueString("foo")},
+			}}),
+			want: ts.ContactInfo{EmailAddress: "foo"},
+		},
 	}
 
 	for _, tt := range tests {
-		var o ts.ContactInfo
-		err := o.FromWire(tt.give)
-		if tt.wantError != "" {
-			if assert.Error(t, err, tt.desc) {
-				assert.Contains(t, err.Error(), tt.wantError)
+		t.Run(tt.desc+"/wire", func(t *testing.T) {
+			var o ts.ContactInfo
+			err := o.FromWire(tt.give)
+			if tt.wantError != "" {
+				if assert.Error(t, err, tt.desc) {
+					assert.Contains(t, err.Error(), tt.wantError)
+				}
+			} else {
+				if assert.NoError(t, err, tt.desc) {
+					assert.Equal(t, tt.want, o)
+				}
 			}
-		} else {
-			if assert.NoError(t, err, tt.desc) {
-				assert.Equal(t, tt.want, o)
+		})
+
+		t.Run(tt.desc+"/streaming", func(t *testing.T) {
+			var (
+				o   ts.ContactInfo
+				buf bytes.Buffer
+			)
+
+			protocol.Binary.Encode(tt.give, &buf)
+
+			sr := protocol.BinaryStreamer.Reader(bytes.NewReader(buf.Bytes()))
+			err := o.Decode(sr)
+			if tt.wantError != "" {
+				if assert.Error(t, err, tt.desc) {
+					assert.Contains(t, err.Error(), tt.wantError)
+				}
+			} else {
+				if assert.NoError(t, err, tt.desc) {
+					assert.Equal(t, tt.want, o)
+				}
 			}
-		}
+
+			_, err = sr.ReadBool()
+			require.Error(t, err)
+			assert.Contains(t, err.Error(), "unexpected EOF", "exepected error for continued reads")
```

Comment:
This seems a bit janky in that it's a random `ReadBool`. I think it might be cleaner to retain a reference to the `bytes.NewReader` and [check that it has no unread bytes](https://pkg.go.dev/bytes#Reader.Len).

Reply:
Done.